File: src/client/lib/game/core/player.dart
import 'package:client/game/core/abstract/game_commander.dart';
import 'package:client/game/core/game.dart';
import 'dart:html';
import 'package:client/game/actors/paddle.dart';
import 'package:client/game/core/abstract/command_router.dart';
import 'package:client/game/core/command.dart';

class Player extends GameCommander
{
  Game game;
  Paddle paddle;

  Player(this.paddle): super()
  {

  }

  run()
  {
    handleInput();
  }

  handleInput()
  {
    if (game.keyboard.isPressed(KeyCode.UP) && !paddle.atGameTopEdge())
    {
      sendCommand(new Command(paddle.ID, 'set_position', [paddle.X, paddle.Y - paddle.speed]));
    }
    else if (game.keyboard.isPressed(KeyCode.DOWN) && !paddle.atGameBottomEdge())
    {
      sendCommand(new Command(paddle.ID, 'set_position', [paddle.X, paddle.Y + paddle.speed]));
    }
  }
}

File: src/game/game_core/player.py
from .game_commander import GameCommander


class Player(GameCommander):
    ID = None

    game = None
    paddle = None

    def __init__(self, player_id, paddle):
        self.ID = player_id
        self.paddle = paddle

    def run(self):
        pass

    def owns_paddle(self, paddle_id):
        return self.paddle.ID == paddle_id

    def set_client(self, cid):
        self.ID = cid
